source/nvda.pyw

```diff
@@ -18,14 +19,27 @@ if getattr(sys, "frozen", None):
 	sys.path.append(sys.prefix)
 	appDir = sys.prefix
 else:
+	# we are running from source
+	# Ensure we are inside the NVDA build system's Python virtual environment.
+	nvdaVenv = os.getenv("NVDA_VENV")
+	virtualEnv = os.getenv("VIRTUAL_ENV")
+	if not nvdaVenv or nvdaVenv != virtualEnv:

```

Comment:
This will make it difficult to do the virtual env setup manually, though perhaps this should be considered beyond scope. I was thinking that setting up or modifying the virtual env manually might be helpful when experimenting with new libraries, avoiding recreating the virtual env each time. On the other hand, recreating the virtual env each time from the requirements file will prevent silly mistakes that lead to drawing incorrect conclusions while experimenting.

Reply:
The only way we could allow this would be that if nvda.pyw is executed outside of the virtual environment that we then execute a subprocess of runnvda or runnvdaw.bat depending on whether python or pythonw was used to originally execute it. But I think that starts getting into some possible recursion danger. I feel it is probably better and safer not to allow executing like this from source anymore, without using runnvda or runnvdaw.